packages/ui/Input/index.tsx

```diff
@@ -0,0 +1,190 @@
+import * as React from "react";
+import styled, { css } from "styled-components";
+
+import { Icon } from "../Icon";
+
+export const Input = ({
+  label,
+  hasError = false,
+  id,
+  supportingLabel,
+  type,
+  value,
+  onChange,
+}: InputProps) => {
+  const randomId = React.useId();
+  const [typeOverride, setTypeOverride] = React.useState("");
+  const inputRef = React.useRef<HTMLInputElement | null>(null);
+
+  const getButton = () => {
+    if (hasError) {
+      return (
+        <StyledIcon>
+          <Icon icon="error" filled color="#b3261e" iconSize={20} />
+        </StyledIcon>
+      );
+    }
+    if (type === "password") {
+      return typeOverride ? (
+        <StyledIcon
+          onClick={() => {
+            setTypeOverride("");
+            inputRef.current?.focus();
+          }}
+        >
+          <Icon
+            icon="visibility_off"
+            color="#49454f"
+            iconSize={23}
+          />
+        </StyledIcon>
+      ) : (
+        <StyledIcon
+          onClick={() => {
+            setTypeOverride("text");
+            inputRef.current?.focus();
+          }}
+        >
+          <Icon
+            icon="visibility"
+            color="#49454f"
+            iconSize={23}
+          />
+        </StyledIcon>
+      );
```

Comment:
```suggestion
return (
<StyledIcon
         onClick={() => {
            setPasswordShown(!passwordShown)
            inputRef.current?.focus()
          }}
        >
          <Icon
            icon={`${passwordShown ? 'visibility_off' : 'visibility'}`}
            color="#49454f"
            iconSize={23}
          />
</StyledIcon>
)
```
This code works great, but i think you overdid it. What would you say we cut this code in half ? ✂️ 
Of course we have to remember to change how we handle Input type with this change

Reply:
Thanks, I did this change, just slightly different on one line, since typescript was mad 😅